SSM/Projectiles/BabyBaconBombProjectile.java

```diff
@@ -0,0 +1,116 @@
+package SSM.Projectiles;
+
+import SSM.Events.SmashDamageEvent;
+import SSM.Utilities.ServerMessageType;
+import SSM.Utilities.Utils;
+import SSM.Utilities.VelocityUtil;
+import net.minecraft.server.v1_8_R3.EntityCreature;
+import net.minecraft.server.v1_8_R3.EnumParticle;
+import org.bukkit.*;
+import org.bukkit.block.Block;
+import org.bukkit.craftbukkit.v1_8_R3.entity.CraftCreature;
+import org.bukkit.entity.Entity;
+import org.bukkit.entity.LivingEntity;
+import org.bukkit.entity.Pig;
+import org.bukkit.entity.Player;
+import org.bukkit.inventory.ItemStack;
+import org.bukkit.scheduler.BukkitRunnable;
+import org.bukkit.util.Vector;
+
+import java.util.HashMap;
+
+public class BabyBaconBombProjectile extends SmashProjectile {
+
+    private int pig_task;
+
+    public BabyBaconBombProjectile(Player firer, String name) {
```

Comment:
You shouldn't hard code values like damage, they should be here in the constructor, including knockback multipliers being set to 0, and hitbox size